src/controller/proxy/controller.go

```diff
@@ -155,6 +155,9 @@ func (c *controller) UseLocalManifest(ctx context.Context, art lib.ArtifactInfo,
 	remoteRepo := getRemoteRepo(art)
 	exist, desc, err := remote.ManifestExist(remoteRepo, getReference(art)) // HEAD
 	if err != nil {
+		if errors.IsRateLimitError(err) && a != nil { // if rate limit, use local if it exists, otherwise return error
```

Comment:
Why do we need to treat 429 as a special case?  Generally, if there's an error when refreshing the cache, and the data exists in the cache and the cache is not expired, we should use the data in the cache